INSTALL.md

```diff
@@ -0,0 +1,323 @@
+Installing From Sources
+=======================
+
+What Do You Need ?
+------------------
+
+To compile Coq yourself, you need:
+
+- [OCaml](https://ocaml.org/) (version >= 4.05.0)
+  (This version of Coq has been tested up to OCaml 4.09.0)
+
+- The [num](https://github.com/ocaml/num) library; note that it is
+  included in the OCaml distribution for OCaml versions < 4.06.0
+
+- The [findlib](http://projects.camlcity.org/projects/findlib.html) library (version >= 1.8.0)
+
+- GNU Make (version >= 3.81)
+
+- a C compiler
+
+- an IEEE-754 compliant architecture with rounding to nearest
+  ties to even as default rounding mode (most architectures
+  should work nowadays)
+
+- for CoqIDE, the
+  [lablgtk3-sourceview3](https://github.com/garrigue/lablgtk) library
+  (version >= 3.0.beta8), and the corresponding GTK 3.x libraries, as
+  of today (gtk+3 >= 3.18 and gtksourceview3 >= 3.18)
+
+The IEEE-754 compliance is required by primitive floating-point
+numbers (`Require Import Floats`). Common sources of incompatibility
+are checked at configure time, preventing compilation. In the,
+unlikely, event an incompatibility remains undetected, using Floats
+would enable to prove False on this architecture.
+
+Note that `num` and `lablgtk3-sourceview3` should be properly
+registered with `findlib/ocamlfind` as Coq's makefile will use it to
+locate the libraries during the build.
+
+Debian / Ubuntu users can get the necessary system packages for
+CoqIDE with:
+
+    $ sudo apt-get install libgtksourceview-3.0-dev
+
+Opam (https://opam.ocaml.org/) is recommended to install OCaml and
+the corresponding packages.
+
+    $ opam install num ocamlfind lablgtk3-sourceview3
```

Comment:
What about including the switch creation phase?

Reply:
Ok, tweaked a bit.